controllers/addresspool_controller.go

```diff
@@ -47,20 +49,27 @@ const (
 // +kubebuilder:rbac:groups=metallb.io,resources=addresspools/status,verbs=get;update;patch
 
 func (r *AddressPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("addresspool", req.NamespacedName)
-	log.Info("Reconciling AddressPool resource")
+	_ = r.Log.WithValues("addresspool", req.NamespacedName)
+	r.Log.Info(fmt.Sprintf("Starting AddressPool reconcile loop for %v", req.NamespacedName))
 
 	instance := &metallbv1alpha1.AddressPool{}
+	defer r.Log.Info(fmt.Sprintf("Finish AddressPool reconcile loop for %v", req.NamespacedName))
+
 	if err := r.Get(ctx, req.NamespacedName, instance); err != nil {
+		if errors.IsNotFound(err) {
+			err = r.resyncMetallbAddressPool(req)
```

Comment:
I believe there's no need to differentiate between del and add.
Let's just always use resync, provided the comments below are applied. We would benefit from having all the logic in one single place.